docs/book/src/architecture/controllers/cluster.md

```diff
@@ -1 +1,45 @@
-# Cluster
+# Cluster Controller
+
+The cluster controller is responsible for keeping the cluster object in sync with the infrastructure cluster object.
+
+Its main responsibilities are:
+
+* Setting an OwnerReference on the infrastructure object.
+* Deleting all related objects in the correct order.
+* Keeping the Cluster object up to date with external objects.
+* Creating a kubeconfig secret for child clusters.
+
+## Contracts
+
+### Infrastructure Provider
+
```

Comment:
We should have a description of what we expect infrastructure providers to do in their reconciliation